streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -873,9 +924,58 @@ private void addStandbyTasks() {
                 checkpointedOffsets.putAll(task.checkpointedOffsets());
             }
         }
-        // finally destroy any remaining suspended tasks
+
+        // destroy any remaining suspended tasks
         removeSuspendedStandbyTasks();
 
+        // create all newly assigned tasks (guard against race condition with other thread via backoff and retry)
+        // -> other thread will call removeSuspendedTasks(); eventually
+        long backoffTimeMs = 50L;
+        while (true) {
```

Comment:
As above. Also, this block of code looks very similar to the one above for active tasks. Is it possible to abstract it enough that they could both share the same code? At least then we can test it once and not duplicate bugs etc